pkg/providers/cloudstack/template.go

```diff
@@ -66,20 +50,14 @@ func getWorkerNodeGroupMachineSpec(clusterSpec v1alpha1.ClusterSpec, machineConf
 		}
 	}
 
-	if len(workerNodeGroupMachineSpecs) == 0 {
-		return workerNodeGroupMachineSpecs, errors.Errorf("getting worker node group configs")
-	}
-
-	return workerNodeGroupMachineSpecs, nil
+	return workerNodeGroupMachineSpecs
 }
 
-func getControlPlaneMachineSpec(clusterSpec v1alpha1.ClusterSpec, machineConfigs map[string]*v1alpha1.CloudStackMachineConfig) (*v1alpha1.CloudStackMachineConfigSpec, error) {
+func getControlPlaneMachineSpec(clusterSpec v1alpha1.ClusterSpec, machineConfigs map[string]*v1alpha1.CloudStackMachineConfig) *v1alpha1.CloudStackMachineConfigSpec {
 	var controlPlaneMachineSpec *v1alpha1.CloudStackMachineConfigSpec
 	if clusterSpec.ControlPlaneConfiguration.MachineGroupRef != nil && machineConfigs[clusterSpec.ControlPlaneConfiguration.MachineGroupRef.Name] != nil {
 		controlPlaneMachineSpec = &machineConfigs[clusterSpec.ControlPlaneConfiguration.MachineGroupRef.Name].Spec
-	} else {
-		return controlPlaneMachineSpec, errors.Errorf("getting MachineGroupRef")
```

Comment:
The cluster config is already being validated [here](https://github.com/aws/eks-anywhere/blob/c98621273ade48012679a66f9bcc62e72f40a49d/pkg/api/v1alpha1/cluster.go#L384), so there is no need for this

Reply:
This validation only checks that it isn't nil, so we could still see an error here if they don't specify a machineGroupRef name. But to resolve this, we just have to be sure to add this to the CloudStack reconciler when we add the additional validations: https://github.com/aws/eks-anywhere/blob/main/pkg/providers/cloudstack/validator.go#L113 , we do the same thing for vsphere

Reply:
Yes, that is the plan.

Additionally, I'm mainly removing this here because then they would writing tests for these cases. This would be wasted effort because we're planning to simplify the `templateBuilder`, such that these functions would no longer be necessary just like with vSphere